### PR TITLE
qa/deepsea: migrate osd config test cases to new YAML format and introduce deepsea/tier4

### PR DESCRIPTION
**Pull request by smithfarm**
The tier numbers express how difficult it is for the tests inside to pass. The current state (as of this PR) is:

* tier0 (deployment smoke test) -> trivially easy to pass
* tier1 (deployment + DS functests) -> easy to pass, low demand on infrastructure
* tier2 (various deployment scenarios: fs, nfs-ganesha, rgw) -> easy to pass, provided infrastructure is cooperating
* tier3 (currently rbd and stage-5) -> tests that are expected to pass but take a little bit longer to run than tier2
* tier4 -> tests that take a long time to run, and/or use lots of resources, or are under development

---

- [ ] merge #229 
- [ ] merge #230 
- [ ] merge #231
- [ ] rebase
- [ ] run tests again

### CODE REVIEW

**Comment by smithfarm** [comment]
@jschmid1 They pass. I'll do another full run on SLE and openSUSE and then merge.

Thanks!

**Comment by smithfarm** [comment]
Test report:

DeepSea RPM:

* `--suite suse` SLE-15 **PENDING**
* `--suite suse` openSUSE Leap 15.0 **PENDING**

DeepSea tip of master branch:

* `--suite suse` SLE-15 **PENDING**
* `--suite suse` openSUSE Leap 15.0 **PENDING**
* `--suite deepsea/tier3` SLE-15 **PENDING**
* `--suite deepsea/tier3` openSUSE Leap 15.0 **PENDING**

File: qa/tasks/salt_manager.py [hunk 1]
```diff
@@ -123,18 +129,39 @@ def all_minions_zypper_status(self):
...
+
+    def cat_salt_minion_confs(self):
+        self.__cat_file_cluster(filename="/etc/salt/minion")
+
     def check_salt_daemons(self):
         self.master_remote.run(args=['sudo', 'salt-key', '-L'])
-        self.master_remote.run(args=[
-            'sudo', 'systemctl', 'status', 'salt-master.service'
-            ])
+        base_cmd = 'sudo systemctl status --full --lines={} {}.service'
```
Comment (kshtsk) [inline qa/tasks/salt_manager.py]: It is better to have embedded function for this purpose:
```
def check_service(self, name):
    def service_status_cmd(remote, service, lines):
         return 'sudo systemctl status --full --lines={} {}.service'.format(lines, service)
    try:
        remote.sh(service_status_cmd(name, 0))
    except CommandFailedError:
        remote.sh(service_status_cmd(name, 100))
self.check_service(self.master_remote, 'salt-master')
for r in self.ctx.cluster.remotes.iterkeys():
    self.check_service(r, 'salt-minion')
.
.
.

```

Comment (smithfarm) [reply qa/tasks/salt_manager.py]: cool. Added to wip-qa-cleanup.

**Comment by smithfarm** [comment]
Tests pass - see https://github.com/SUSE/ceph/pull/232#issuecomment-438299613

**Pull request merged**